libraries/WebServer/src/WebServer.cpp

```diff
@@ -796,3 +803,13 @@ String WebServer::_responseCodeToString(int code) {
     default:  return F("");
   }
 }
+
+#if SOC_WIFI_SUPPORTED
+  bool ON_STA_FILTER(WebServer &server) {
+    return WiFi.STA.localIP() == server.client().localIP();
+  }
+
+  bool ON_AP_FILTER(WebServer &server) {
+    return WiFi.AP.localIP() == server.client().localIP();
```

Comment:
```cpp
return WiFi.AP.hasIP() && WiFi.AP.localIP() == server.client().localIP();
```

Else it could return IP 0 which will also match with the client IP being 0

Reply:
Done with [e92f4e6](https://github.com/espressif/arduino-esp32/pull/9842/commits/e92f4e640c4aebb47235f5d290e39ea283957b5b)